hydro_lang/src/keyed_stream.rs

```diff
@@ -74,7 +74,43 @@ where
     }
 }
 
+/// Helper function to create a KeyedStream from a transformed underlying stream,
+/// updating the metadata to reflect keyed stream type conversions.
+fn keyed_stream_with_updated_metadata<'a, K, U, L, B, O, R>(
+    underlying: Stream<(K, U), L, B, NoOrder, R>,
+) -> KeyedStream<K, U, L, B, O, R>
+where
+    L: Location<'a>,
+{
+    let is_bounded = std::any::type_name::<B>().contains("Bounded");
```

Comment:
Can we use a trait here?

Reply:
Hmm isn't this new code in this PR? We don't do anything like this elsewhere (or if it was accidentally approved in a previous PR that was a mistake). I think we should fix this before merging. `type_name` is very tricky to get right and should not be used directly...

Reply:
addressed